InOrder_PreOrder.py

```diff
@@ -0,0 +1,67 @@
+  ---------------------- Construct Binary Tree Inorder Preorder------------------------------------------------   
+# Time Complexity : O(N) N is the number of nodes.
+# Space Complexity : O(N) N is the number of nodes
+# Did this code successfully run on Leetcode : Yes
+# Any problem you faced while coding this : No
+# 
+# Here preorder is the order of nodes with left , curr, and right node. Inorder is the order where root node will be first. 
+# In this sense, the nodes on the left indexes of element are its left nodes and right are its right nodes. Also the preorder will have all the root nodes in the first.
+# So recursively , we will create a root with preorder index element and find the index of the element in inorder and assign the left elements to root.left and right elements to root.right.
+# and we will call recursive function again with new preorder and inorder.
+
+class Solution:
+    def buildTree(self, preorder: List[int], inorder: List[int]) -> TreeNode:
+        if not preorder or not inorder:
+            return None
+        
+        def dfs(preorder, inorder):
+            if not preorder or not inorder:
+                return None
+            
+
+            index = inorder.index(preorder[0])
```

Comment:
Is this operation O(1) always?
If not, then your time complexity would not be O(n) using recursion.